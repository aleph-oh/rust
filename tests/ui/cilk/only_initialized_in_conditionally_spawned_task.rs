#![feature(cilk)]
// Tests that when we sync conditionally spawned tasks, the liveness analysis correctly considers
// the variables uninitialized.

// compile-flags: -C panic=abort
// no-prefer-dynamic

fn f() -> usize {
    let y;
    if true {
        cilk_spawn { y = 1; };
    }
    cilk_sync;
    y + 1
    //~^ ERROR used binding `y` is possibly-uninitialized [E0381]
}

fn main() {
}
