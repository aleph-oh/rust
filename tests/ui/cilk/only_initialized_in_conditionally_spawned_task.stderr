error[E0381]: used binding `y` is possibly-uninitialized
  --> $DIR/only_initialized_in_conditionally_spawned_task.rs:14:5
   |
LL |     let y;
   |         - binding declared here but left uninitialized
LL |     if true {
LL |         cilk_spawn { y = 1; };
   |                      ----- binding initialized here in some conditions
...
LL |     y + 1
   |     ^ `y` used here but it is possibly-uninitialized

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0381`.
